scvi/models/utils.py

```diff
@@ -44,3 +44,10 @@ def batch(batch_size, label):
 
     batch_size = x.size(0)
     return torch.cat([batch(batch_size, i) for i in range(y_dim)])
+
+
+def logexpsum(logx, logy):
```

Comment:
I believe `torch.logsumexp` implements this

Reply:
I'm assuming you're referring to this: https://pytorch.org/docs/master/generated/torch.logsumexp.html

That is a possible option (although the syntax is slightly different).